ViewManagement/Initialize.lua

```diff
@@ -220,22 +222,59 @@ local function SetupBankView()
     ResetPositions()
   end
 
+  local function GetSelectedBankTab(entity)
+    if type(entity) == "string" or entity == nil then -- Character bank
+      return characterTab
+    elseif type(entity) == "number" then -- Warband bank
+      return warbandTab
+    end
+  end
+
   addonTable.CallbackRegistry:RegisterCallback("ResetFramePositions", function()
     ResetPositions()
   end)
 
-  addonTable.CallbackRegistry:RegisterCallback("BankToggle", function(_, characterName)
-    characterName = characterName or Syndicator.API.GetCurrentCharacter()
-    bankView:SetShown(characterName ~= bankView.Character.lastCharacter or not bankView:IsShown())
-    bankView:UpdateViewToCharacter(characterName)
+  addonTable.CallbackRegistry:RegisterCallback("BankToggle", function(_, entity, subView)
+
+    local currentTab = bankView:GetCurrentTab()
+    local selectedTab = GetSelectedBankTab(entity)
+
+    if currentTab == selectedTab then
+      -- Toggle bank view
+      if selectedTab == characterTab then -- Character bank
+        local characterName = entity or Syndicator.API.GetCurrentCharacter()
+        bankView:SetShown(characterName ~= bankView.Character.lastCharacter or not bankView:IsShown())
+        bankView:UpdateViewToCharacter(characterName)
+      elseif selectedTab == warbandTab then -- Warband bank
+        subView = subView or addonTable.Config.Get(addonTable.Config.Options.WARBAND_CURRENT_TAB)
+        bankView:SetShown(not bankView:IsShown())
+        bankView:UpdateViewToWarband(entity, subView)
+      end
+    else
```

Comment:
This should be simplified to close the bank if the toggle for the current view is used.

Reply:
Should be all done now. Feel free to verify 😄